shundayo131/HarmonyMaker
# Colin/setup front and back

# Linked issue/ticket
HM-5

# Description
This PR implements the initial TypeScript React frontend setup integrated with Express backend:

1. Frontend Configuration:
   - Set up TypeScript with ES6 target
   - Configure webpack for dev/prod environments
   - Implement React component structure
   - Set up development server on port 3033

2. Backend Integration:
   - Configure Express server on port 8080
   - Set up and tested static file serving
   - test the test middleware
   - Verify route handling

3. Development Environment:
   - Add concurrent to front and backend server 
   - Configure nodemon with TypeScript support
   - Implement ts-node for development
   - Organize component file structure

# Reproduction steps
1. Check frontend server setup:
   * Run `npm start` and verify frontend runs on port 3033
   * Confirm React components render correctly from 3033/

2. Validate backend functionality:
   * Run `npm run start:server` and verify backend on port 8080
   * Test `/api/test` endpoint responds correctly
   * Confirm static files are served properly at 8080/test.css

3. Test concurrent execution:
   * Run `npm run dev`
   * Verify both servers start
   * Confirm frontend can communicate with backend at 3033/

# Checklist
* [x] I've followed the Contributing guidelines
* [x] I've titled my PR according to the Conventional Commits spec
* [x] TypeScript files compile without errors
* [x] Webpack configuration works for dev environment 
* [x] Backend routes and middleware are tested
* [x] Concurrent server execution is verified
